File: 0001_两数之和/main.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


class Solution:
    def twoSum(self, nums: List[int], target: int):
        hash_map = dict()
        for index, x in enumerate(nums):
            if target - x in hash_map:
                return [index, hash_map[target - x]]
            hash_map[x] = index

File: 0001_两数之和/main.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i, x := range nums{
        v, ok := m[target - x]
        if ok{
            return []int{v, i}
        }
        m[x] = i
    }
    return nil
}
